dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.ImageGeneration;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI.CustomClient;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.ImageGeneration;
+
+public class AzureOpenAIImageGeneration : OpenAIClientBase, IImageGeneration
+{
+    /// <summary>
+    /// Constants of success status
+    /// </summary>
+    private const string SucceededStatus = "SUCCEEDED";
+
+    /// <summary>
+    /// Maximum retry times for the Azure Dall-E API
+    /// </summary>
+    private readonly int _maxRetryCount;
+
+    /// <summary>
+    /// Default retry wait time in seconds
+    /// </summary>
+    private readonly int _defaultRetryAfter;
+
+    /// <summary>
+    /// Azure OpenAI REST API endpoint
+    /// </summary>
+    private readonly string _endpoint;
+
+    /// <summary>
+    /// Azure OpenAI API key
+    /// </summary>
+    private readonly string _apiKey;
+
+    /// <summary>
+    /// Create a new instance of Azure OpenAI image generation service
+    /// </summary>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="logger">Application logger</param>
+    /// <param name="maxRetryCount">Maximum number of retries for HTTP requests.</param>
```

Comment:
Please update the comment for the maxRetryCount parameter to clearly indicate that it's related to the maximum number of attempts to made to retrieve an image generation operation result, rather than to resilience-related retries. 
For example: "Maximum number of attempts to retrieve the image generation operation result."